lua/wire/client/cl_wirelib.lua

```diff
@@ -34,181 +34,79 @@ end
 local beam_mat = mats["tripmine_laser"]
 local beamhi_mat = mats["Models/effects/comball_tape"]
 
+local lastrender, scroll, shouldblink = 0, 0, false
 function Wire_Render(ent)
-	if (not ent:IsValid()) then return end
-	if (Wire_DisableWireRender == 1) then return end
-
 	if (Wire_DisableWireRender == 0) then
-		local path_count = ent:GetNetworkedBeamInt("wpn_count") or 0
-		if (path_count <= 0) then return end
-
-		local w,f = math.modf(CurTime()*WIRE_BLINKS_PER_SECOND)
-		local blink = nil
-		if (f < 0.5) then
-			blink = ent:GetNetworkedBeamString("BlinkWire")
-		end
-
-		for i = 1,path_count do
-			local path_name = ent:GetNetworkedBeamString("wpn_" .. i)
-			if (blink ~= path_name) then
-				local net_name = "wp_"..path_name
-				local len = ent:GetNetworkedBeamInt(net_name) or 0
-
-				if (len > 0) then
+		local wires = ent.WirePaths
+		if wires and next(wires) then
+
+			local t = CurTime()
+			if lastrender ~= t then
+				local w, f = math.modf(t*WIRE_BLINKS_PER_SECOND)
+				shouldblink = f < 0.5
+				scroll = t*WIRE_SCROLL_SPEED
+				lastrender = t
+			end
 
-					local width = ent:GetNetworkedBeamFloat(net_name .. "_width")
-					if width > 0 then
+			local blink = shouldblink and ent:GetNetworkedBeamString("BlinkWire")
```

Comment:
`"BlinkWire"` is now the only place in Wiremod that uses `{Get,Set}NetworkedBeam*`. Can you kill it off (by just replacing it with `{Get,Set}NWString`) and then remove `beam_netvars.lua`?